modules/dnn/test/test_model.cpp

```diff
@@ -738,6 +914,178 @@ TEST_P(Test_Model, TextDetectionByEAST)
     );
 }
 
+TEST_P(Test_Model, FaceDetectionBySSD)
+{
+    if (target == DNN_TARGET_OPENCL_FP16)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_OPENCL_FP16);
+
+    // Weight
+    std::string face_detection_weight_path = _tf("opencv_face_detector.caffemodel", false);
+    std::string face_detection_config_path = _tf("opencv_face_detector.prototxt");
+
+    // Ground Truth
+    std::string image_path = findDataFile("gpu/lbpcascade/er.png");
+
+    std::vector<Rect> gt_boxes;
+    gt_boxes.push_back(Rect(408, 84, 34, 49));
+    gt_boxes.push_back(Rect(142, 256, 32, 43));
+    gt_boxes.push_back(Rect(118, 46, 34, 45));
+    gt_boxes.push_back(Rect(340, 34, 29, 42));
+    gt_boxes.push_back(Rect(65, 123, 37, 43));
+    gt_boxes.push_back(Rect(259, 228, 32, 43));
+
+    std::vector<float> gt_confidences;
+    gt_confidences.push_back(0.672344f);
+    gt_confidences.push_back(0.665494f);
+    gt_confidences.push_back(0.646890f);
+    gt_confidences.push_back(0.388749f);
+    gt_confidences.push_back(0.283762f);
+    gt_confidences.push_back(0.256311f);
+
+    // Threshold
+    const double iouDiff = 0.9;
+    const double scoreDiff = 2e-4;
+
+    // Run Test
+    testFaceDetectionModelBySSD(
+        face_detection_weight_path,
+        face_detection_config_path,
+        image_path,
+        gt_boxes,
+        gt_confidences,
+        iouDiff,
+        scoreDiff
+    );
+}
+
+TEST_P(Test_Model, FaceDetectionByYN)
+{
+    if (target == DNN_TARGET_OPENCL_FP16)
+        applyTestTag(CV_TEST_TAG_DNN_SKIP_OPENCL_FP16);
+
+    // Weight
+    std::string face_detection_weight_path = _tf("onnx/models/yunet-202109.onnx", false);
```

Comment:
[A fix](https://github.com/opencv/opencv/pull/21607) is on-they-way. A new version of YuNet should be used instead after the fix is merged.

Reply:
Okay, I'll fix to load new model after that pull request merged.